src/riak_kv_wm_timeseries_listkeys.erl

```diff
@@ -77,164 +74,122 @@ service_available(RD, Ctx = #ctx{riak = RiakProps}) ->
-    end.
-
--spec check_permissions(#wm_reqdata{}, #ctx{}) -> ?CB_RV_SPEC.
-check_permissions(RD, Ctx = #ctx{security = undefined}) ->
-    {true, RD, Ctx};
-check_permissions(RD, Ctx = #ctx{security = Security,
-                                 table = Table}) ->
-    case riak_core_security:check_permission(
-           {riak_kv_ts_util:api_call_to_perm(listkeys), Table}, Security) of
-        {false, Error, _} ->
-            handle_error(
-              {not_permitted, utf8_to_binary(Error)}, RD, Ctx);
-        _ ->
-            {true, RD, Ctx}
+        {error, ErrorMsg} ->
+            {ErrorMsg, RD, Ctx};
+        {insecure, Halt, Resp} ->
+            {Halt, Resp, Ctx}
     end.
 
+-spec forbidden(#wm_reqdata{}, #ctx{}) -> cb_rv_spec(boolean()).
+forbidden(RD, Ctx) ->
+   Result = riak_kv_wm_utils:is_forbidden(RD),
+    {Result, RD, Ctx}.
 
--spec malformed_request(#wm_reqdata{}, #ctx{}) -> ?CB_RV_SPEC.
-malformed_request(RD, Ctx = #ctx{api_version = "v1"}) ->
-    {false, RD, Ctx};
-malformed_request(RD, Ctx = #ctx{api_version = UnsupportedVersion}) ->
-    handle_error({unsupported_version, UnsupportedVersion}, RD, Ctx).
-
-
--spec allowed_methods(#wm_reqdata{}, #ctx{}) ->
-    {[atom()], #wm_reqdata{}, #ctx{}}.
+-spec allowed_methods(#wm_reqdata{}, #ctx{}) -> cb_rv_spec([atom()]).
 %% @doc Get the list of methods this resource supports.
 allowed_methods(RD, Ctx) ->
     {['GET'], RD, Ctx}.
 
-
--spec resource_exists(#wm_reqdata{}, #ctx{}) ->
-                             {boolean(), #wm_reqdata{}, #ctx{}}.
-resource_exists(RD, #ctx{table = Table} = Ctx) ->
-    Mod = riak_ql_ddl:make_module_name(Table),
-    case catch Mod:get_ddl() of
-        {_, {undef, _}} ->
-            handle_error({no_such_table, Table}, RD, Ctx);
-        _ ->
-            {true, RD, Ctx}
-    end.
-
+-spec resource_exists(#wm_reqdata{}, #ctx{}) -> cb_rv_spec(boolean()).
+resource_exists(RD, #ctx{mod=Mod} = Ctx) ->
+    {riak_kv_wm_ts_util:table_module_exists(Mod), RD, Ctx}.
 
 -spec encodings_provided(#wm_reqdata{}, #ctx{}) ->
-                                {[{Encoding::string(), Producer::function()}],
-                                 #wm_reqdata{}, #ctx{}}.
+                                cb_rv_spec([{Encoding::string(), Producer::function()}]).
 %% @doc List the encodings available for representing this resource.
 %%      "identity" and "gzip" are available.
 encodings_provided(RD, Ctx) ->
     {riak_kv_wm_utils:default_encodings(), RD, Ctx}.
 
 -spec content_types_provided(#wm_reqdata{}, #ctx{}) ->
-                                    {[{ContentType::string(), Producer::atom()}],
-                                     #wm_reqdata{}, #ctx{}}.
+                                    cb_rv_spec([{ContentType::string(), Producer::atom()}]).
 %% @doc List the content types available for representing this resource.
 content_types_provided(RD, Ctx) ->
-    {[{"application/json", produce_doc_body}], RD, Ctx}.
-
+      {[{"text/html", produce_doc_body}], RD, Ctx}.
```

Comment:
Is "text/html" suitable here? Streamed data are supposed to be consumed by some machinery and not rendered in a browser.